src/App/App.tsx

```diff
@@ -43,14 +50,13 @@ export default function App() {
         isMovieInfoOpen={isMovieInfoOpen}
         movie={movie}
       />
-      <Main
-        onMovieModalOpen={handleMovieModalOpen}
-        onDeleteModalOpen={handleDeleteModalOpen}
-        onMovieInfoOpen={handleMovieInfoOpen}
-        isLoading={isLoading}
-        movies={movies}
-        loadingError={error}
-      />
+      <CardContext.Provider value={cardContextValue}>
+        <Main
+          isLoading={isLoading}
+          movies={movies}
+          loadingError={error}
+        />
+      </CardContext.Provider>
```

Comment:
Let's create a separate component with context and modals.

Reply:
Created a ModalProvider component, imported and placed an AppContext into it.